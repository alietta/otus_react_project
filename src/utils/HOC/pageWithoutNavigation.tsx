import React from "react";
import { DarkMode, Container, useTheme } from "sancho";
import { pageStyle } from "./pageStyle";

export const pageWithoutNavigation = <Props extends object>(
  Component: React.ComponentType<Props>
) => (props: Props) => {
  const theme = useTheme();
  return (
    <div css={{ width: "100vw", height: "100vh", background: theme.colors.background.default}}>
      <Container css={{ maxWidth: 1024, height: '100%' }}>
        <Component {...props} />
      </Container>
    </div>
  );
};
