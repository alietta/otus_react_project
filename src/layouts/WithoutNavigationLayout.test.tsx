import React from "react";
import { mount } from "enzyme";
import { pageWithoutNavigation } from "./pageWithoutNavigation";

describe("HOC pageWithoutNavigation", () => {
  interface ComponentProps {
    name: string;
  }
  const Component: React.FC<ComponentProps> = ({ name }) => <h1>{name}</h1>;
  const WrappedComponent = pageWithoutNavigation(Component);
  it("render", () => {
    const element = mount(<WrappedComponent name="Anny" />);
    expect(element.html()).toMatchInlineSnapshot(
      `"<div class=\\"css-1xfup8k\\"><div class=\\"css-1jywssl\\"><h1>Anny</h1></div></div>"`
    );
  });
});
