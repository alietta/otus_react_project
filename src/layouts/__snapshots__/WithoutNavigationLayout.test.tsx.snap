// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithoutNavigationLayout render 1`] = `"<div class=\\"css-73e4gw-WithoutNavigationLayout\\"><div class=\\"css-mvnpm9-WithoutNavigationLayout\\"><h1>Test</h1></div></div>"`;
